### PR TITLE
Fix auth log msgs and server issue with content-less payload

### PR DESCRIPTION
**Pull request by yakman2020**
Prerequisites for encrypt

### CODE REVIEW

**Comment by yakman2020** [comment]
@Microsoft/omi-devs

File: Unix/http/httpauth.c [hunk 1]
```diff
@@ -1552,15 +1555,50 @@ MI_Boolean IsClientAuthorized(_In_ Http_SR_SocketData * handler)
...
+                    // Apparently we were just authorising the connection so far and the request is 
+                    //  yet to come. Treat this in the same way as a continue except succeed
+                    
+                    handler->httpErrorCode = HTTP_ERROR_CODE_OK;
+                    if (output_token.length != 0)
+                    {
+                        auth_response = _BuildAuthResponse(protocol_p, handler->httpErrorCode, &output_token, &response_len);
+                        if (auth_response == NULL)
+                        {
+                            // Problem : 2do complain into trace file
```
Comment (jeffaco) [inline Unix/http/httpauth.c]: Should do this now, prior to checkin, lest it be forgotten.

File: Unix/http/httpclientauth.c [hunk 2]
```diff
@@ -681,6 +687,292 @@ static char *_BuildBasicAuthHeader(_In_ struct _HttpClient_SR_SocketData *self,
...
+            trace_Socket_ConnectionClosed(handler);
+            return FALSE;
+
+        }
+        else if ((int)sent < 0)
+        {
+            switch (SSL_get_error(handler->ssl, sent))
+            {
+
+             // These do not happen. We havfe already drained the socket
```
Comment (jeffaco) [inline Unix/http/httpclientauth.c]: Fix spelling: `havfe`

File: Unix/http/httpclientauth.c [hunk 3]
```diff
@@ -681,6 +687,292 @@ static char *_BuildBasicAuthHeader(_In_ struct _HttpClient_SR_SocketData *self,
...
+    };
+    
+    const gss_OID_set_desc mechset_krb5 = { 2, (gss_OID) mechset_krb5_elems };
+    
+    OM_uint32 maj_stat = 0;
+    OM_uint32 min_stat = 0;
+    gss_ctx_id_t context_hdl = GSS_C_NO_CONTEXT;
+    gss_buffer_desc input_token, output_token;
+    gss_OID_set mechset = NULL;
+    //OM_uint32 ret_flags = 0;
```
Comment (jeffaco) [inline Unix/http/httpclientauth.c]: Remove if not needed

File: Unix/http/httpauth.c (hunk 1 above)
Comment (yakman2020) [inline Unix/http/httpauth.c]: done

File: Unix/http/httpclientauth.c (hunk 2 above)
Comment (yakman2020) [inline Unix/http/httpclientauth.c]: done

**Pull request merged**